### PR TITLE
Update branch cutting instructions.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

Add instructions for creating dev tag.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: RELEASING.md
Comment (tmjd) [inline RELEASING.md:38]: Why do we need this commit? Why not just put a tag on the most recent commit?

Comment (fasaxc) [reply RELEASING.md:38]: That is explained in the paragraph immediately above ;-)

**Review by tmjd — APPROVED** [review]
LGTM, lets wait for Dan to review before merging.

**Review by danudey — APPROVED** [review]
lgtm

**Pull request merged**
